drivers/usb/device/usb_dc_sam0.c

```diff
@@ -181,6 +204,7 @@ int usb_dc_attach(void)
 
 	while (GCLK->STATUS.bit.SYNCBUSY) {
 	}
+#endif /* !MCLK */
 
 	/* Configure */
 	regs->CTRLA.bit.SWRST = 1;
```

Comment:
Don't the interrupts 2 and 3 also need to be connected (TRCPT0 and TRCPT1).  It looked like you have them in the DTS already.

Reply:
Good catch!
The `samples/subsys/usb/cdc_acm` sample is still not working, I'll have to investigate this further.